app/models/user_collection.rb

```diff
@@ -3,4 +3,6 @@ class UserCollection < ApplicationRecord
   belongs_to :user
   has_many :article_collections
   has_many :articles, through: :article_collections
+  validates :title, presence: true
+  validates_uniqueness_of :title, scope: :user_id, case_sensitive: false
```

Comment:
Great work implementing a catch so a user can't add duplicate articles in the same collection!